cloudformation/memsub-promotions-lambdas-cf.yaml

```diff
@@ -0,0 +1,253 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Description: Membership/Subscription promotions tool lambda functions
+Parameters:
+  DEVPromotionsTableStreamARN:
+    Description: ARN of the MembershipSub-Promotions-DEV DynamoDB stream to trigger
+      the lambda function which updates the MembershipSub-PromoCode-View-DEV table
+    Type: String
+    Default: ''
+  UATPromotionsTableStreamARN:
+    Description: ARN of the MembershipSub-Promotions-UAT DynamoDB stream to trigger
+      the lambda function which updates the MembershipSub-PromoCode-View-UAT table
+    Type: String
+    Default: ''
+  PRODPromotionsTableStreamARN:
+    Description: ARN of the MembershipSub-Promotions-PROD DynamoDB stream to trigger
+      the lambda function which updates the MembershipSub-PromoCode-View-PROD table
+    Type: String
+    Default: ''
+Resources:
+  MembershipSubPromotionsToolRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Statement:
+        - Effect: Allow
+          Principal:
+            Service:
+            - lambda.amazonaws.com
+          Action:
+          - sts:AssumeRole
+      Path: /
+      Policies:
+      - PolicyName: MembershipSub-Promotions-Tool-Policy
+        PolicyDocument:
+          Version: '2012-10-17'
+          Statement:
+          - Effect: Allow
+            Action:
+            - lambda:InvokeFunction
+            - logs:CreateLogGroup
+            - logs:CreateLogStream
+            - logs:PutLogEvents
+            Resource: '*'
```

Comment:
We should limit these permissions to the appropriate resource

Reply:
(particularly for lambda:InvokeFunction)

Reply:
I actually don't understand why lambda:InvokeFunction is needed, the Lambdas don't invoke one another, instead they trigger off Dynamo stream listeners and CloudWatch alarms. Likewise I also don't understand why AssumeRole is needed here.

As mentioned this is a separation of the old CFN not an integral part of my change. Let's discuss when we do the alarms discussion below.